server/src/main/java/org/apache/iotdb/db/metadata/MManager.java

```diff
@@ -1901,6 +1901,12 @@ public MNode getSeriesSchemasAndReadLockDevice(InsertPlan plan)
           if (!config.isAutoCreateSchemaEnabled()) {
             throw new PathNotExistException(deviceId + PATH_SEPARATOR + measurementList[i]);
           } else {
+            // check measurementId syntax
+            if (measurementList[i].contains(".")
+                && !(measurementList[i].startsWith("\"") && measurementList[i].endsWith("\""))) {
+              throw new MetadataException(
+                  String.format("%s is an illegal measurementId", measurementList[i]));
+            }
```

Comment:
I see a method in MTree called `checkTimeseries`. Should we use that?

Reply:
That's may be better encapsulation. I've updated my code.

Here's some explanation. In this problem, once the path is transformed from PartialPath to String, like root.sg.d.a.b with measurementId a.b, the regex is unable to check the problem. Only when the path is in format of name list, String[] nodeNames, can the problem be detected. 

The string concat operation on path may exist widely in iotdb's code. That's why I suggested above, filter the illegal syntax before create query plan.